clojure.c

```diff
@@ -0,0 +1,163 @@
+/**
+ *   Copyright (c) 2015, Miloslav Nenadál <nenadalm@gmail.com>
+ *
+ *   This source code is released for free distribution under the terms of the
+ *   GNU General Public License.
+ *
+ *   This module contains code for generating tags for the Clojure language.
+ */
+
+#include <string.h>
+
+#include "general.h"
+
+#include "parse.h"
+#include "read.h"
+#include "vstring.h"
+#include "entry.h"
+
+typedef enum {
+	K_FUNCTION,
+	K_NAMESPACE
+} clojureKind;
+
+static kindOption ClojureKinds[] = {
+	{TRUE, 'f', "function", "functions"},
+	{TRUE, 'n', "namespace", "namespaces"}
+};
+
+static vString *CurrentNamespace;
+
+static int isNamespace (const unsigned char *strp)
+{
+	return strncmp (++strp, "ns", 2) == 0 && isspace (strp[2]);
+}
+
+static int isFunction (const unsigned char *strp)
+{
+	return strncmp (++strp, "defn", 4) == 0 && isspace (strp[4]);
+}
+
+static int isQuote (const unsigned char *strp)
+{
+	return strncmp (++strp, "quote", 5) == 0 && isspace (strp[5]);
+}
+
+static void functionName (vString * const name, const unsigned char *dbp)
+{
+	const unsigned char *p;
+
+	if (*dbp == '\'')
+		dbp++;
+	else if (*dbp == '(' && isQuote (dbp))
+	{
+		dbp += 7;
+		while (isspace (*dbp))
+			dbp++;
+	}
```

Comment:
Hum… wait, what is this supposed to match?  I mean, stop me if I miss something, but from `findClojureTags()` loop, you only match with `isNamespace()` (`(ns `) or `isFunction()` (`(defn `).  Then, in either case you skip the word (`ns` or `defn`) and the following whitespaces, and read a name with `functionName()`.  `functionName()` will match [`^(?:'|\(quote +)([^() \t]+)`](http://regexper.com/#%2F^%28%3F%3A%27|\\%28quote%20%2B%29%28%5B^%28%29%20\t%5D%2B%29%2F) -- e.g. a function name prefixed by either nothing, `'` or `(quote `.

So, IIUC it allows to match constructs like `(ns 'somename)` or `(ns (quote someothername))`.  If that's it, there should be a test for them :)